modules/storages/lib/open_project/storages/engine.rb

```diff
@@ -62,6 +69,47 @@ class Engine < ::Rails::Engine
               ::Storages::ManageNextcloudIntegrationEventsJob.debounce
             end
           end
+
+          OpenProject::Notifications.subscribe(
+            OpenProject::Events::OAUTH_CLIENT_TOKEN_CREATED
+          ) do |payload|
+            if payload[:integration_type] == 'Storages::Storage'
+              ::Storages::ManageNextcloudIntegrationEventsJob.debounce
+            end
+          end
+          OpenProject::Notifications.subscribe(
+            OpenProject::Events::ROLE_CREATED
+          ) do |payload|
+            if payload[:permissions].intersect?(PERMISSIONS)
+              ::Storages::ManageNextcloudIntegrationEventsJob.debounce
+            end
+          end
+          OpenProject::Notifications.subscribe(
+            OpenProject::Events::ROLE_UPDATED
+          ) do |payload|
+            if payload[:permissions_diff].intersect?(PERMISSIONS)
+              ::Storages::ManageNextcloudIntegrationEventsJob.debounce
+            end
+          end
+          OpenProject::Notifications.subscribe(
+            OpenProject::Events::ROLE_DESTROYED
+          ) do |payload|
+            if payload[:permissions].intersect?(PERMISSIONS)
+              ::Storages::ManageNextcloudIntegrationEventsJob.debounce
+            end
+          end
```

Comment:
I am surprised about the choice of events in here. My understanding of the intend is to update the permissions on Nextcloud in lockstep with the changes in OpenProject.

If my understanding is correct then ROLE_CREATE does not need to be listed here. It means a role has been created. But the creation of a role does not mean that a permission granted to the user is changed. Only if the newly created role is granted as part of the membership will there be a change in permission. As such, I don't think that listening for this change is necessary.

Reply:
Yes, you are absolutely right!
I unsubscribed from PROJECT_CREATED as well for similar reasons.

Reply:
Is is true that non members (i.e. a user in a public project) cannot have permissions in Nextcloud (I didn't check). Only if that is true should the `PROJECT_CREATED` be removed as otherwise the project created might be public which would then grant permissions.